lib/core/request.js

```diff
@@ -164,10 +164,21 @@ class Request {
         processHeader(this, headers[i], headers[i + 1])
       }
     } else if (headers && typeof headers === 'object') {
-      const keys = Object.keys(headers)
-      for (let i = 0; i < keys.length; i++) {
-        const key = keys[i]
-        processHeader(this, key, headers[key])
+      if (headers[Symbol.iterator]) {
+        for (const header of headers) {
+          if (!Array.isArray(header) || header.length !== 2) {
+            throw new InvalidArgumentError('headers must be in key-value pair format')
+          }
+          const key = header[0]
+          const value = header[1]
+          processHeader(this, key, value)
+        }
+      } else {
+        const keys = Object.keys(headers)
+        for (let i = 0; i < keys.length; i++) {
```

Comment:
would it be healthier?
```suggestion
        for (const key of keys) {
```

Reply:
It's a bit debatable.  
The main purpose of this PR is to introduce a new feature. Making changes to code that isn't directly related feels like it may violate the principle of separation of concerns.

Reply:
I agree with you, it will be more performant to use it this way, can we try it if it does not pose any problem?

Reply:
The for...of loop assigns each element of the array directly to the key variable and you don't need to check the indexes as you traverse the array. This makes the code more readable and reduces the risk of errors.

With a for loop, you have to check the index and increment the index at each step. This type of loop can lead to potential sources of errors. Also, if the number of elements of the array is very large, keys.length must be calculated at each loop step, which negatively affects performance.

Reply:
I did some research, I hope it will be useful to us, you can see it in my comment above.

Reply:
That's reasonable. Let's add it.

Reply:
Thank you very much for your contribution